File: java/v2/src/main/java/com/api/v2/persons/domain/PersonRepository.java
package com.api.v2.persons.domain;

import org.springframework.data.mongodb.repository.Query;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import reactor.core.publisher.Mono;

import java.util.UUID;

public interface PersonRepository extends ReactiveMongoRepository<Person, UUID> {

    @Query("{ 'ssn': ?0 }")
    Mono<Person> findBySsn(String ssn);

    @Query("{ 'email': ?0 }")
    Mono<Person> findByEmail(String email);

}

File: kotlin/v3/src/main/kotlin/com/api/v3/persons/domain/PersonRepository.kt
package com.api.v3.persons.domain

import org.springframework.data.mongodb.repository.Query
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import java.util.UUID

interface PersonRepository: CoroutineCrudRepository<Person, UUID> {

    @Query("{ 'ssn': ?0 }")
    fun findBySsn(ssn: String): Person?

    @Query("{ 'email': ?0 }")
    fun findByEmail(email: String): Person?

}